app/components/Search/mazemapAutocomplete.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+
+import React, { Component, type ComponentType } from 'react';
+import { debounce } from 'lodash';
+import { stripHtmlTags } from './utils.js';
+import "node_modules/mazemap/mazemap.min.css";
+import * as Mazemap from 'mazemap';
+
+type InjectedProps = {
+  mazemapSearch: (query: string) => Promise<*>,
+  meta: any,
+};
+
+type State = {
+  searching: boolean,
+  result: Array<Object>,
+};
+
+function mazemapAutocomplete<Props>({
+  WrappedComponent,
+}: {
+  WrappedComponent: ComponentType<Props>,
+}) {
+  const displayName =
+    WrappedComponent.displayName || WrappedComponent.name || 'Unknown';
+
+  const mySearch = new Mazemap.Search.SearchController({
+    campusid: 1,
+    rows: 10,
+    withpois: true,
+    withbuilding: false,
+    withtype: false,
+    withcampus: false,
+    resultsFormat: 'geojson',
+  });
+
+  return class extends Component<InjectedProps & Props, State> {
+    static displayName = `Autocomplete(${displayName})`;
+
+    state = {
+      searching: false,
+      result: [], //first result fetched on mount
+    };
+
+    _isMounted = false;
+
+    componentDidMount() {
+      this._isMounted = true;
+      if (this.props.meta.initial) {
+        Mazemap.Data.getPoi(this.props.meta.initial.value).then((poi) =>
+          this.setState({
+            result: [
+              {
+                label:
+                  poi.properties.title + ', ' + poi.properties.buildingName,
+                value: this.props.meta.initial.value,
+              },
```

Comment:
I would create a helper method that creates this result `{object}`. It can take the title, name, and value as params and return this. Then it can be used within `mySearch.search()` _(which should be called something else)_.